### DISCUSSION TITLE
Power dissipation of the 3.3V regulator

### DISCUSSION THREAD

**Discussion by longissor** [gab-k/RP2040-Decoder#30] [Q&A]
Hello,

First, thank you for sharing your project.

I built it and in the process of testing and I am running into few issues. The main one is the heat generated by one of the component.
I programmed it all good.
I set it up with a motor and my Roco Z21.
With address 3, I am able to control the motor (speed and direction). So all good so far.

But then I noticed the project switch off and back on and off etc... and one of the UC is getting super hot. Hot enough to burn my finger. It is small, but I suspect it is the 3.3V regulator.
The project uses between 60mA and 70mA. I read that the RP2040 uses about 24 to 30mA when running
My DCC is at around 18V
The dissipation should be (18-3.3)*x30 = 441mW
But the spec of the regulator shows a power dissipation of 300mW

I am right in my assumptions? Is it why my project is getting hot and shut down?
Did you have this issue at some point in your dev?

This other issue is none of the functions work. I cannot switch a LED. But the heat issue about prevent me to investigate it.

Many thank for your help.

**Comment by gab-k** [comment]
Yes the regulator is a bad design, i wanted to change it a long time ago already, just didn't get around to it.

My suggestion is replacing the regulator with a bigger form factor one that can handle the power dissipation.

**Reply by longissor** [reply]
Thanks
I will try to reduce the voltage n my DCC system to 10-12 in order t continue my test.
I will let you know if I find a regulator that still allow for a small enough solution

**Reply by gab-k** [reply]
you could for example use a regulator in a SOT-223 or DPAK package

**Reply by longissor** [reply]
Yes, If I manage to validate everything else, I will go back to the regulator issue. I already have few references in SOT-233 and TO-252 package in mind. 
They are much bigger and I will have to fit some of the components on the other side to keep the project small enought for HO scale. And may be move from 2 layers to 4 layers. It will make the project more complex and costly to manufacture.

to be continued....
